### DISCUSSION TITLE
Adding an ImageInput::valid_file overload that takes an IOProxy

### DISCUSSION THREAD

**Discussion by jessey-git** [OpenImageIO/oiio#3820] [General]
Over the past little while I've been helping transition most of Blender's image loading/saving code to OIIO. Most of the work was made possible with the expanded IOProxy work done for 2.4. Thank you very much for that!

One piece that didn't quite fit into the existing machinery was the lack of an overload to `ImageInput::valid_file` that also takes in an IOProxy. Some format loaders do this privately but it's not a part of the main API.

I've prototyped it a bit but wanted to check-in here before continuing with all the remaining formats.

Would you be willing to accept a PR that adds such an overload for `ImageInput::valid_file`? And, if so, do you prefer an actual overloaded method or do you prefer to add a default parameter to the existing method with the IOProxy defaulted to nullptr?

**Comment by lgritz** [comment]
That makes sense, and I would encourage such a PR.

With a second parameter, if you pass the IOProxy, you don't need the filename, and vice versa, right?  That seems awkward. So I think I prefer the separately overloaded approach, rather than the additional optional parameter.

Please note that changing the virtual functions in ImageInput is an ABI incompatibility, so we can merge into master but can't backport to the 2.4 release branch. It wouldn't be in a supported release series until later this year (2.5, which would be allowed to change ABI relative to 2.4, is expected to be released by Sept 1).
